gha-cache/src/api.rs

```diff
@@ -580,10 +583,7 @@ impl Api {
                         if response.ok {
                             Ok(offset)
                         } else {
-                            Err(Error::ApiError {
-                                status: StatusCode::BAD_REQUEST,
-                                info: ApiErrorInfo::Unstructured(Bytes::new()),
-                            })
+                            Err(Error::ApiErrorNotOk)
```

Comment:
Hm, using ApiError seemed okay/appropriate here?

Reply:
I waffled on this a couple times; I changed it because it felt awkward to synthesize a fake error code (the error message would be kind of confusing.) Can revert this back though.